flink-libraries/flink-table/src/main/scala/org/apache/flink/table/sources/TableSource.scala

```diff
@@ -19,22 +19,23 @@
 package org.apache.flink.table.sources
 
 import org.apache.flink.api.common.typeinfo.TypeInformation
+import org.apache.flink.table.api.TableEnvironment
 
-/** Defines an external table by providing schema information, i.e., field names and types.
+/** Defines an external table by providing schema information and used to produce a
+  * [[org.apache.flink.api.scala.DataSet]] or [[org.apache.flink.streaming.api.scala.DataStream]].
+  * Schema information consists of a data type, field names, and corresponding indices of
+  * these names in the data type.
+  *
+  * To define a TableSource one need to implement [[TableSource#getReturnType]]. In this case
+  * field names and field indices are derived from the returned type.
+  *
+  * In case if custom field names are required one need to additionally implement
```

Comment:
I am not sure about this. I've checked it with [Grammarly](grammarly.com) and it does not complain about "In case if", but complains about the "in case of".